operator/src/main/java/oracle/kubernetes/operator/DomainStatusUpdater.java

```diff
@@ -1029,7 +1028,24 @@ private boolean isNonClusteredServer(String serverName) {
       }
 
       private boolean isRunning(@Nonnull String serverName) {
-        return RUNNING_STATE.equals(getRunningState(serverName));
+        return RUNNING_STATE.equals(getRunningState(serverName)) && isRollCompleteFor(serverName);
+      }
+
+      private boolean isRollCompleteFor(@Nonnull String serverName) {
+        return isAdminServer(serverName) || isNotMarkedForRoll(serverName);
```

Comment:
Why is roll always complete for admin server?

Reply:
We don't roll the admin server. If it needs to be replaced, we do so immediately. We roll the managed server pods so as never to have too many offline at once. Since you are the second person to ask, it seems that a comment or better names are needed.